nnsvs/gen.py

```diff
@@ -81,22 +82,49 @@ def predict_timelag(device, labels, timelag_model, timelag_in_scaler, timelag_ou
 
     # Run model
     x = torch.from_numpy(timelag_linguistic_features).unsqueeze(0).to(device)
-    y = timelag_model(x, [x.shape[1]]).squeeze(0).cpu()
 
-    # De-normalization and rounding
-    lag = np.round(timelag_out_scaler.inverse_transform(y.data.numpy()))
+    # Run model
+    if timelag_model.prediction_type == "probabilistic":
```

Comment:
Same as above.

Reply:
I changed to use  predictin_type() of BaseModel in the latest commit.